roles/ansible/tower/config-ansible-tower/tasks/install.yml

```diff
@@ -3,9 +3,10 @@
 - block: # become: True
 
   - name: "install epel-release"
-    package:
+    yum:
       name: "{{ ansible_tower_epel_download_url }}"
       state: present
+      disable_gpg_check: yes
```

Comment:
Let's make this a configuration through the inventory, and default it to `no`.

Reply:
We need to add an additional task then to import the GPG key, otherwise, this will fail all the time. I agree to move the value to a variable in the inventory but we should default it to `yes`. I don't expect people to import EPEL GPG key manually. While using EPEL normally this is imported when is prompted to the user, but not as a regular task they do on their hosts.

Reply:
you can also just add the gpg key https://docs.ansible.com/ansible/latest/modules/rpm_key_module.html

Reply:
Yeah, the GPG keys are here:
https://dl.fedoraproject.org/pub/epel/

Perhaps update the `defaults/main.yml` to something like:
```
ansible_tower_epel_download_url: https://dl.fedoraproject.org/pub/epel
ansible_tower_epel_rpm: epel-release-latest-7.noarch.rpm
ansible_tower_epel_gpg: RPM-GPG-KEY-EPEL-7
```

... then use those to compose the call to `yum` and `rpm_key`.

Reply:
If we know which EPEL version the user is using yes, but we don't know that as the user can provide it's own rpm URL to override the default one provided by the role.

Reply:
Are you guys suggesting to force the user to provide the GPG file? I'm fine with that.

Reply:
@makentenza well, in this case it allows the user the option to specify it ... but otherwise it will just use the default value(s).

Something like (note the `omit` default):

```
- rpm_key:
    key: "{{ ansible_tower_epel_download_url }}/{{ ansible_tower_epel_gpg }}"
  when:
    - ansible_tower_epel_gpg is defined
    - ansible_tower_epel_gpg|trim != ""

- yum:
    name: "{{ ansible_tower_epel_download_url }}/{{ ansible_tower_epel_rpm }}"
    state: present
    disable_gpg_check: "{{ ansible_tower_epel_disable_gpg | default(omit) }}"
```

Reply:
If we add default values I'm not fully understanding this:

```
- rpm_key:
    key: "{{ ansible_tower_epel_download_url }}/{{ ansible_tower_epel_gpg }}"
  when:
    - ansible_tower_epel_gpg is defined
    - ansible_tower_epel_gpg|trim != ""
```

Since `ansible_tower_epel_gpg` will be defined and I don't expect the user adding the variable with no value to the inventory just to override the defailt one with an empty string? Users don't do that normally.

If the GPG is not installed then this will fail:

```
- yum:
    name: "{{ ansible_tower_epel_download_url }}/{{ ansible_tower_epel_rpm }}"
    state: present
    disable_gpg_check: "{{ ansible_tower_epel_disable_gpg | default(omit) }}"
```

So I'd suggest to check if the GPG has been provided or not to make a decision there. Let me come with some changes and then we can discuss further?

Reply:
Hi @makentenza - not sure it's worth the effort. The original comment as just to address the fact that's not a good idea to hard-code / permanently disable a check that's meant to ensure validity of the source. However, this thread is now too much. The suggestion I provided in the previous post would work for your use-case - i.e.: it will have the GPG file and it will install it before it tries to install the EPEL RPM (which will then succeed because it has the GPG key) + it would provide the user with the option to supply their own values, or completely disable the GPG check if so desire. 

Anyway - I can just accept your PR as-is, and we can look to address further at a later point.

Reply:
Sure, that works for me.